frontend/src/game.js

```diff
@@ -2,31 +2,223 @@ import * as PIXI from 'pixi.js';
 import { Hex_grid } from './grid';
 import { Unit } from './unit';
 
+
 export class Game {
-    // players_num - number of players (for the future)
-    // app - PIXI aplication, used for this game
-    // grid - grid for this game (Hex_grid)
-    constructor(players_num = 2, grid_params = {}) {
-        this.players_num = players_num;
+    constructor() {
+        // app - PIXI aplication, used for this game
         this.app = null;
+        // grid - grid for this game (Hex_grid)
         this.grid = null;
 
+        // Websocket we use to contact with server
+        this.socket = null;
+
+        // players_num - number of players (for the future)
+        this.players_num = 2;
+        // my_num - number of player using this client
+        this.my_num = 1;
+
+        // array of game states
+        this.STATES = {
+            PREPARE: 'prepare',
+            MOVE_FROM: 'move_from',
+            MOVE_TO: 'move_to',
+            ATTACK: 'attack',
+            WAIT: 'wait',
+            END: 'end'
+        };
+        this.state = this.STATES.PREPARE;
+        // current player's move and attack
+        this.cur_move = {from: null, to: null};
+        this.cur_attack = null;
+
         // graphic constants
         this.BACKGROUND_COLOR = 0xd6b609;
     }
 
-    // needed, because constructor cannot be async
-    async start() {
+    // (needed, because constructor cannot be async)
+    start() {
+        this.create_socket();
+    }
+
+    // create websocket to receive and send messages
+    // private
+    create_socket() {
+        this.socket = new WebSocket("ws://127.0.0.1:8088/ws/");
+
+        this.socket.onopen = function(e) {
+            console.log("[open] Connection established");
+        };
+
+        this.socket.onmessage = this.process_message.bind(this);
+
+        this.socket.onclose = function(event) {
+            if (event.wasClean) {
+                console.log(`[close] Connection closed cleanly, code=${event.code} reason=${event.reason}`);
+            } else {
+                // e.g. server process killed or network down
+                // event.code is usually 1006 in this case
+                console.log('[close] Connection died');
+            }
+        };
+
+        this.socket.onerror = function(error) {
+            console.log(`[error] ${error.message}`);
+        };
+    }
+
+
+    // States operations
+    // check there is correct state
+    // private
+    check_state(...states) {
+        if (! states.includes(this.state)) {
+            throw new Error(`Incorrect game state: ${this.state}; ${states} expected`);
+        }
+    }
+
+    //private
+    change_state(state) {
+        this.state = state;
+    }
+
+
+    // Game cycle
+    // process messages from the server
+    // private
+    process_message() {
+        let data = JSON.parse(event.data);
+        console.log(data);
+        // TODO: we need to wait here all previous actions done
+        switch (data.cmd) {
```

Comment:
I think it would be better to use map from `cmd` to function, so we can just write something like
`cmd_map[data.cmd]();`

With this approach our code can be more flexible.

Reply:
So we need several functions - for each command, yea?
Like this:
```
cmd_map['field'] = function() {
    ... // some code
}
```
?

Reply:
Yes.

Reply:
done!